includes/class-wc-ajax.php

```diff
@@ -1994,146 +1877,45 @@ public static function load_variations() {
 
 		check_ajax_referer( 'load-variations', 'security' );
 
-		// Check permissions again and make sure we have what we need
-		if ( ! current_user_can( 'edit_products' ) || empty( $_POST['product_id'] ) || empty( $_POST['attributes'] ) ) {
+		if ( ! current_user_can( 'edit_products' ) || empty( $_POST['product_id'] ) ) {
 			die( -1 );
 		}
 
+		// Set $post global so its available, like within the admin screens
 		global $post;
 
-		$product_id = absint( $_POST['product_id'] );
-		$post       = get_post( $product_id ); // Set $post global so its available like within the admin screens
-		$per_page   = ! empty( $_POST['per_page'] ) ? absint( $_POST['per_page'] ) : 10;
-		$page       = ! empty( $_POST['page'] ) ? absint( $_POST['page'] ) : 1;
-
-		// Get attributes
-		$attributes        = array();
-		$posted_attributes = wp_unslash( $_POST['attributes'] );
-
-		foreach ( $posted_attributes as $key => $value ) {
-			$attributes[ $key ] = array_map( 'wc_clean', $value );
-		}
-
-		// Get tax classes
-		$tax_classes           = WC_Tax::get_tax_classes();
-		$tax_class_options     = array();
-		$tax_class_options[''] = __( 'Standard', 'woocommerce' );
-
-		if ( ! empty( $tax_classes ) ) {
-			foreach ( $tax_classes as $class ) {
-				$tax_class_options[ sanitize_title( $class ) ] = esc_attr( $class );
-			}
-		}
-
-		// Set backorder options
-		$backorder_options = array(
-			'no'     => __( 'Do not allow', 'woocommerce' ),
-			'notify' => __( 'Allow, but notify customer', 'woocommerce' ),
-			'yes'    => __( 'Allow', 'woocommerce' ),
-		);
-
-		// set stock status options
-		$stock_status_options = array(
-			'instock'    => __( 'In stock', 'woocommerce' ),
-			'outofstock' => __( 'Out of stock', 'woocommerce' ),
-		);
-
-		$parent_data = array(
-			'id'                   => $product_id,
-			'attributes'           => $attributes,
-			'tax_class_options'    => $tax_class_options,
-			'sku'                  => get_post_meta( $product_id, '_sku', true ),
-			'weight'               => wc_format_localized_decimal( get_post_meta( $product_id, '_weight', true ) ),
-			'length'               => wc_format_localized_decimal( get_post_meta( $product_id, '_length', true ) ),
-			'width'                => wc_format_localized_decimal( get_post_meta( $product_id, '_width', true ) ),
-			'height'               => wc_format_localized_decimal( get_post_meta( $product_id, '_height', true ) ),
-			'tax_class'            => get_post_meta( $product_id, '_tax_class', true ),
-			'backorder_options'    => $backorder_options,
-			'stock_status_options' => $stock_status_options,
+		$loop           = 0;
+		$product_id     = absint( $_POST['product_id'] );
+		$post           = get_post( $product_id );
+		$product_object = wc_get_product( $product_id );
+		$per_page       = ! empty( $_POST['per_page'] ) ? absint( $_POST['per_page'] ) : 10;
+		$page           = ! empty( $_POST['page'] ) ? absint( $_POST['page'] ) : 1;
+		$variations     = get_posts(
```

Comment:
We should try to use the new `wc_get_products` here. If it's missing a parameter to filter with, it should be easy to add.